packages/mui-material/src/utils/getChildRef.js

```diff
@@ -1,8 +1,16 @@
 import * as React from 'react';
 
 export default function getChildRef(child) {
-  if (!child || !React.isValidElement(child)) {
-    return null;
+  if (process.env.NODE_ENV !== 'production') {
+    if (!React.isValidElement(child)) {
```

Comment:
I guess we could use [`React.Children.only`](https://react.dev/reference/react/Children#children-only) although se'd lose the custom error message.